Assets/Scripts/UI/InGameUI/MenuLeft/MenuLeft.cs

```diff
@@ -45,11 +41,11 @@ public void OpenMenu(string menuName)
         }
     }
 
-    public void CloseMenu()
+    public void CloseCurrentlyMenu()
     {
         if (CurrentlyOpen != null)
         {
-            CurrentlyOpen.GetComponent<SlideAnimation>().Hide();
+            CurrentlyOpen.GetComponent<IGameMenu>().Close();
```

Comment:
iGameMenu is an interface, not a Monobehavior. Not sure why GetComponent is useful here...

Reply:
We need only the IGameMenu functions. I implemented IGameMenu to ConstructorMenu, and OrderMenu MonoBehaviour scripts. We need an interface because menus can use different animations. For example if we want to deactivate all menus we can call all menus easily via IGameMenu interface. But if we don't use interface, we will face this kind of an issue:
```
public void CloseAll()
        {
            if (menu.GetComponent<SlideAnimation>() != null)
            {
                menu.GetComponent<SlideAnimation>().Hide();
            }
            else if (menu.GetComponent<FadeAnimation>() != null)
            {
                menu.GetComponent<FadeAnimation>().FadeOut();
            }
            .
            .
            .
        }
```